### PR TITLE
CDAP-14401 auto create pubsub topic if it doesnt exist

### PR DESCRIPTION
**Pull request by albertshau**
Enhanced the pubsub sink to create the topic if it doesn't exist.
This will prevent users from running lengthy pipelines just to
realize at the end that the topic needs to be created.

### CODE REVIEW

File: src/main/java/co/cask/gcp/publisher/GooglePublisher.java [hunk 1]
```diff
@@ -66,8 +74,33 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
...
-  public void prepareRun(BatchSinkContext context) {
+  public void prepareRun(BatchSinkContext context) throws IOException {
     config.validate();
+
+    TopicAdminSettings.Builder topicAdminSettings = TopicAdminSettings.newBuilder();
+    String serviceAccountPath = config.getServiceAccountFilePath();
+    if (serviceAccountPath != null) {
+      topicAdminSettings.setCredentialsProvider(() -> GCPUtils.loadServiceAccountCredentials(serviceAccountPath));
+    }
+    TopicAdminClient topicAdminClient = TopicAdminClient.create(topicAdminSettings.build());
```
Comment (rohitsinha54) [inline src/main/java/co/cask/gcp/publisher/GooglePublisher.java]: We should call close() on the topicAdminClient or use with try-with-resource. The class documentation says:

```
 * <p>Note: close() needs to be called on the topicAdminClient object to clean up resources such as
 * threads. In the example above, try-with-resources is used, which automatically calls close().
```

File: src/main/java/co/cask/gcp/publisher/GooglePublisher.java [hunk 2]
```diff
@@ -66,8 +74,33 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
...
+
+    String projectId = config.getProject();
+    ProjectTopicName projectTopicName = ProjectTopicName.of(projectId, config.topic);
+
+    try {
+      topicAdminClient.getTopic(projectTopicName);
+    } catch (NotFoundException e) {
```
Comment (rohitsinha54) [inline src/main/java/co/cask/gcp/publisher/GooglePublisher.java]: Is NotFoundException guaranteed to be  be thrown if it does not exists?

File: src/main/java/co/cask/gcp/publisher/GooglePublisher.java (hunk 1 above)
Comment (albertshau) [reply src/main/java/co/cask/gcp/publisher/GooglePublisher.java]: good call, fixed

File: src/main/java/co/cask/gcp/publisher/GooglePublisher.java (hunk 2 above)
Comment (albertshau) [reply src/main/java/co/cask/gcp/publisher/GooglePublisher.java]: from my experiments, this is how it behaves. Doesn't seem to return null. The client looks like an auto-generated one, so it maps pretty directly to the rest endpoint.

**Comment by rohitsinha54** [comment]
LGTM.

**Pull request merged**